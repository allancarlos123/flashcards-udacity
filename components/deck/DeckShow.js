import _ from 'lodash'
import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { connect } from 'react-redux';
import { removeDeck, deckFetch} from './../../actions/decks'
import { Ionicons } from '@expo/vector-icons';

class DeckShow extends Component {
  static navigationOptions = ({ navigation }) => ({
    headerLeft: <Ionicons name='ios-arrow-back' size={30} onPress={() => navigation.navigate('DeckList')} style={styles.backButton} />,
  });

  componentDidMount = () => {
    const { title } = this.props.navigation.state.params
    this.props.deckFetch(title)
  };

  onPressDeleteDeck = () => {
    const { title } = this.props.deck

    this.props.deleteDeck(title, () => {
      this.props.navigation.navigate('DeckList')
    })
  }

  onPressStartQuiz = () => {
    this.props.navigation.navigate('Quiz')
  }

  render() {
    const { title, questions } = this.props.deck

    return (
      <View style={styles.container}>
        <Text>{title} ({_.size(questions)})</Text>

        <Button
          onPress={this.onPressDeleteDeck}
          title="Delete Deck"
          color="#000"
        />
        <Button
          onPress={this.onPressStartQuiz}
          title="Start Quiz"
          color="red"
        />
        <Button
          onPress={() => this.props.navigation.navigate('QuizCreate')}
          title="Add Quiz"
          color="purple"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 10
  },
  backButton: {
    marginLeft: 15
  }
});

mapStateToProps = ({deck}) => {
  return {
    deck: deck.deck
  }
}

mapDispatchToProps = dispatch => {
  return {
    deckFetch: id => dispatch(deckFetch(id)),
    deleteDeck: (deck, callback) => dispatch(removeDeck(deck, callback))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DeckShow);