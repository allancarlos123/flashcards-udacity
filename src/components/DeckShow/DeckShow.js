import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import PrimaryButton from "../Buttons/Primary";

const DeckShow = ({
  deckEmpty,
  onPressDeleteDeck,
  onPressStartQuiz,
  navigation
}) => (
  <View>
    <PrimaryButton
      title="Start"
      disabled={deckEmpty}
      onPress={() => onPressStartQuiz()}
    />

    <PrimaryButton
      title="Add card"
      onPress={() => navigation.navigate("CreateQuiz")}
    />

    <PrimaryButton title="Delete deck" onPress={() => onPressDeleteDeck()} />
  </View>
);

DeckShow.propTypes = {
  deckEmpty: PropTypes.bool.isRequired,
  onPressStartQuiz: PropTypes.func.isRequired,
  onPressDeleteDeck: PropTypes.func.isRequired
};

export default DeckShow;
