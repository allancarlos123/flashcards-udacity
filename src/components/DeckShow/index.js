import DeckShow from "./DeckShow";

export default DeckShow