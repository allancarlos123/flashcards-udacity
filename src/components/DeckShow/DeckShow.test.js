import React from 'react'
import { create } from 'react-test-renderer'

import DeckShow from './'

describe('DeckShow', () => {
  const onPressDeleteDeck = jest.fn();
  const onPressStartQuiz = jest.fn();
  const navigation = jest.fn();
  const tree = create(
    <DeckShow
      deckEmpty={false}
      onPressDeleteDeck={onPressDeleteDeck}
      onPressStartQuiz={onPressStartQuiz}
      navigation={navigation}
    />
  );

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});