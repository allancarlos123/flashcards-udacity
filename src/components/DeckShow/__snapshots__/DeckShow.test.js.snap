// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckShow renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderColor": "#007AFF",
            "borderRadius": 6,
            "borderWidth": 1,
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#ed4e5f",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#ed4e5f",
            },
            Object {
              "paddingVertical": 12,
            },
          ]
        }
      >
        Start
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ed4e5f",
            "height": "100%",
            "opacity": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "paddingVertical": 12,
              },
              undefined,
              undefined,
            ]
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderColor": "#007AFF",
            "borderRadius": 6,
            "borderWidth": 1,
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#ed4e5f",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#ed4e5f",
            },
            Object {
              "paddingVertical": 12,
            },
          ]
        }
      >
        Add card
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ed4e5f",
            "height": "100%",
            "opacity": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "paddingVertical": 12,
              },
              undefined,
              undefined,
            ]
          }
        >
          Add card
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "borderColor": "#007AFF",
            "borderRadius": 6,
            "borderWidth": 1,
            "overflow": "hidden",
          },
          Object {
            "borderColor": "#ed4e5f",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#ed4e5f",
            },
            Object {
              "paddingVertical": 12,
            },
          ]
        }
      >
        Delete deck
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ed4e5f",
            "height": "100%",
            "opacity": 0,
            "position": "absolute",
            "width": "100%",
          }
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "paddingVertical": 12,
              },
              undefined,
              undefined,
            ]
          }
        >
          Delete deck
        </Text>
      </View>
    </View>
  </View>
</View>
`;
